geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSUnionIntegrationTest.java

```diff
@@ -56,95 +62,185 @@ public void sunionErrors_givenTooFewArguments() {
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKeyDifferentSlot, secondSetMembers);
+
+    assertThatThrownBy(() -> jedis.sunion(setKey1, setKeyDifferentSlot))
+        .hasMessageContaining(ERROR_DIFFERENT_SLOTS);
+  }
+
+
+  @Test
+  public void sunion_withDifferentKeyTypeAndTwoSetKeys_returnsWrongTypeError() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String diffKey = "{tag1}diffKey";
+    jedis.set(diffKey, "dong");
+    assertThatThrownBy(() -> jedis.sunion(diffKey, setKey1, setKey2))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
 
-    jedis.sadd("{tag1}newEmpty", "born2die");
-    jedis.srem("{tag1}newEmpty", "born2die");
-    Set<String> yetAnotherResultSet = jedis.sunion("{tag1}set2", "{tag1}newEmpty");
-    assertThat(yetAnotherResultSet).containsExactlyInAnyOrder(secondSet);
+  @Test
+  public void sunion_withTwoSetKeysAndDifferentKeyType_returnsWrongTypeError() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String diffKey = "{tag1}diffKey";
+    jedis.set(diffKey, "dong");
+    assertThatThrownBy(() -> jedis.sunion(setKey1, setKey2, diffKey))
+        .hasMessageContaining(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void ensureSetConsistency_whenRunningConcurrently() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String[] unionMembers = {"one", "two", "three", "four", "five", "linux", "peach"};
+
+    final AtomicReference<Set<String>> sunionResultReference = new AtomicReference<>();
+    new ConcurrentLoopingThreads(1000,
+        i -> jedis.srem(setKey2, secondSetMembers),
+        i -> sunionResultReference.set(jedis.sunion(setKey1, setKey2)))
+            .runWithAction(() -> {
+              assertThat(sunionResultReference).satisfiesAnyOf(
+                  sunionResult -> assertThat(sunionResult.get())
+                      .containsExactlyInAnyOrder(setMembers1),
+                  sunionResult -> assertThat(sunionResult.get())
+                      .containsExactlyInAnyOrder(unionMembers));
+              jedis.sadd(setKey2, unionMembers);
+            });
   }
 
   @Test
   public void testSUnionStore() {
     String[] firstSet = new String[] {"pear", "apple", "plum", "orange", "peach"};
     String[] secondSet = new String[] {"apple", "microsoft", "linux", "peach"};
     String[] thirdSet = new String[] {"luigi", "bowser", "peach", "mario"};
-    jedis.sadd("{tag1}set1", firstSet);
-    jedis.sadd("{tag1}set2", secondSet);
-    jedis.sadd("{tag1}set3", thirdSet);
+    jedis.sadd("{user1}set1", firstSet);
+    jedis.sadd("{user1}set2", secondSet);
+    jedis.sadd("{user1}set3", thirdSet);
```

Comment:
These (and the other similar keys in this class) should be replaced with the `setKey1`, `setKey2` constants, and the tag on the third key (and other non-constant key names) should be replaced with "{tag1}" to be consistent with the changes from GEODE-9932.